daemon/job_actual_ip.go

```diff
@@ -0,0 +1,67 @@
+package daemon
+
+import (
+	"context"
+	"log"
+	"net/netip"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/core"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/network"
+)
+
+func insightsUntilSuccess(ctx context.Context, api core.InsightsAPI) (core.Insights, error) {
+	for i := 0; ; i++ {
+		select {
```

Comment:
[small] instead of double-select, you could also check `ctx.Err()` at the beginning of the  loop and return if it was cancelled. This way you can have just one select in the loop, maybe it would be easier to follow :shrug:, if not then you can ignore this comment